mlos_bench/mlos_bench/tests/storage/sql/fixtures.py

```diff
@@ -107,120 +105,99 @@ def mixed_numerics_exp_storage(
-        tunables = opt.suggest()
-        for repeat_j in range(CONFIG_TRIAL_REPEAT_COUNT):
-            trial = exp.new_trial(
-                tunables=tunables.copy(),
-                config={
-                    "trial_number": config_i * CONFIG_TRIAL_REPEAT_COUNT + repeat_j + 1,
-                    **{
-                        f"opt_{key}_{i}": val
-                        for (i, opt_target) in enumerate(exp.opt_targets.items())
-                        for (key, val) in zip(["target", "direction"], opt_target)
-                    },
-                },
-            )
-            if exp.tunables:
-                assert trial.tunable_config_id == config_i + 1
-            else:
-                assert trial.tunable_config_id == 1
-            if tunable_name:
-                tunable_value = float(tunables.get_tunable(tunable_name)[0].numerical_value)
-                tunable_value_norm = base_score * (tunable_value - tunable_min) / tunable_range
-            else:
-                tunable_value_norm = 0
-            timestamp = datetime.now(UTC)
-            trial.update_telemetry(
-                status=Status.RUNNING,
-                timestamp=timestamp,
-                metrics=[
-                    (timestamp, "some-metric", tunable_value_norm + random() / 100),
-                ],
-            )
-            trial.update(
-                Status.SUCCEEDED,
-                timestamp,
-                metrics={
-                    # Give some variance on the score.
-                    # And some influence from the tunable value.
-                    "score": tunable_value_norm
-                    + random() / 100
-                },
-            )
-    return exp
-
 
-@pytest.fixture
-def exp_storage_with_trials(exp_storage: SqlStorage.Experiment) -> SqlStorage.Experiment:
-    """Test fixture for Experiment using in-memory SQLite3 storage."""
-    return _dummy_run_exp(exp_storage, tunable_name="kernel_sched_latency_ns")
-
-
-@pytest.fixture
-def exp_no_tunables_storage_with_trials(
-    exp_no_tunables_storage: SqlStorage.Experiment,
-) -> SqlStorage.Experiment:
-    """Test fixture for Experiment using in-memory SQLite3 storage."""
-    assert not exp_no_tunables_storage.tunables
-    return _dummy_run_exp(exp_no_tunables_storage, tunable_name=None)
+    scheduler = SyncScheduler(
+        # All config values can be overridden from global config
+        config={
+            "experiment_id": exp.experiment_id,
+            "trial_id": exp.trial_id,
+            "config_id": -1,
+            "trial_config_repeat_count": CONFIG_TRIAL_REPEAT_COUNT,
+            "max_trials": CONFIG_COUNT * CONFIG_TRIAL_REPEAT_COUNT,
+        },
+        global_config={},
+        environment=env,
+        optimizer=opt,
+        storage=storage,
+        root_env_config=exp.root_env_config,
+    )
 
+    # Add some trial data to that experiment by "running" it.
+    with scheduler:
+        scheduler.start()
+        scheduler.teardown()
```

Comment:
This basically replaces all of the prior dummy run code, but also limits the metadata we've injected into the Trial, so we need to adjust some of the expected test output checks as well.